packages/gatsby-plugin-netlify-cms/src/gatsby-browser.js

```diff
@@ -1,15 +1,33 @@
 /* global __PATH_PREFIX__ */
-import netlifyIdentityWidget from "netlify-identity-widget"
 
-exports.onInitialClientRender = (_, { enableIdentityWidget = true, publicPath = `admin` }) => {
-  if (enableIdentityWidget) {
-    netlifyIdentityWidget.on(`init`, user => {
-      if (!user) {
-        netlifyIdentityWidget.on(`login`, () => {
-          document.location.href = `${__PATH_PREFIX__}/${publicPath}/`
+// Taken from https://github.com/netlify/netlify-identity-widget
+const routes = /(confirmation|invite|recovery|email_change)_token=([^&]+)/
+const errorRoute = /error=access_denied&error_description=403/
+const accessTokenRoute = /access_token=/
+
+exports.onInitialClientRender = (
```

Comment:
Is this whole block anything we want to test for? I'd think adding some tests would be really valuable here, even if "just" unit tests!

Reply:
Yeah, that would be nice. Not sure how tho, since it's a dynamic import things become a bit more tricky. Are there any similar tests a can look at to get going?

Reply:
Instant bug if the source this was copied from changes. Perhaps Netlify Identity links should include a flag query like `netlify_identity_redirect=true`. Opened an issue to discuss: https://github.com/netlify/netlify-identity-widget/issues/157